topdown/http.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package topdown
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/topdown/builtins"
+)
+
+// DefaultHTTPRequestTimeoutSec Default timeout value for http client
+const DefaultHTTPRequestTimeoutSec = 5
+
+var allowedKeysMap map[string]bool
+var client *http.Client
+
+func builtinHTTPReq(bctx BuiltinContext, a ast.Value) (ast.Value, error) {
+	switch a.(type) {
+	case ast.Object:
+		obj, ok := a.(ast.Object)
+		if !ok {
+			return nil, fmt.Errorf("type assertion error")
+		}
+
+		return evaluateHTTPRequest(obj, bctx)
+	default:
+		return nil, builtins.NewOperandTypeErr(1, a, ast.ObjectTypeName)
+	}
+
+}
+
+func init() {
+	createHTTPClient()
+	createAllowedKeysMap()
+	RegisterFunctionalBuiltin1WithCtxt(ast.HTTPReqBuiltin.Name, builtinHTTPReq)
+}
+
+// createHTTPClient creates a HTTP client
+func createHTTPClient() {
+	var timeout int
+
+	timeoutDuration := os.Getenv("HTTP_REQUEST_TIMEOUT_SEC")
+
+	if timeoutDuration == "" {
+		timeout = DefaultHTTPRequestTimeoutSec
+	} else {
+		timeout, _ = strconv.Atoi(timeoutDuration)
+	}
+
+	client = &http.Client{
+		Timeout: time.Duration(timeout) * time.Second,
+	}
+}
+
+// createAllowedKeysMap creates a map of the keys allowed by the builtin
+func createAllowedKeysMap() {
```

Comment:
See comment below about validation. We can define these globally and skip this initialization.